lib/I18n.php

```diff
@@ -193,10 +196,14 @@ public static function loadTranslations()
     public static function getAvailableLanguages()
     {
         if (count(self::$_availableLanguages) == 0) {
-            $i18n = dir(self::_getPath());
-            while (false !== ($file = $i18n->read())) {
-                if (preg_match('/^([a-z]{2,3}).json$/', $file, $match) === 1) {
-                    self::$_availableLanguages[] = $match[1];
+            self::$_availableLanguages[] = 'en'; // en.json is not part of the release archive
+            $languageIterator            = new AppendIterator();
+            $languageIterator->append(new GlobIterator(self::_getPath('??.json')));
+            $languageIterator->append(new GlobIterator(self::_getPath('???.json'))); // for jbo
+            foreach ($languageIterator as $file) {
+                $language = $file->getBasename('.json');
+                if ($language != 'en') {
```

Comment:
This seems to ensure en does not end up two times. Okay likely has no bad effect, but yeah. IN a potential unit test that would be another test case.

As said, feel free to do tests, if you want, in another PR.

Reply:
Correct, the issue is that in the git repo (and therefore during testing) the en.json file is present. It only got introduced for crowdin, as that requires a base translation file to copy for new languages, which is why we used to have logic that artificially added the "en" entry to the list. In the release the file is excluded via gitattributes, as it should not get read by anything (waste of a disk read and parsing CPU-cycles, as there is nothing to do).

Reg. test: Could copy all the translations into a temp dir, change PUBLIC_PATH to point to it, then do one call to getAvailableLanguages() with the file present and one when removed, comparing the two results. It is doable, if a little convoluted. I'll raise another PR later on.

Reply:
Yeah that test idea sounds fine (It's more an integration test, but that's okay, if mocking the dependencies [in this case the file system] is not possible).